lib/logstash/inputs/heartbeat.rb

```diff
@@ -38,34 +43,64 @@ class LogStash::Inputs::Heartbeat < LogStash::Inputs::Threadable
   # This is typically used only for testing purposes.
   config :count, :validate => :number, :default => -1
 
+  # Select the type of sequence, deprecating the 'epoch' and 'sequence' values in 'message'.
+  #
+  # If you set this to `epoch` then this plugin will use the current
+  # timestamp in unix timestamp (which is by definition, UTC).
+  #
+  # If you set this to `sequence` then this plugin will send a sequence of
+  # numbers beginning at 0 and incrementing each interval.
+  #
+  # If you set this to 'none' then no field is created
+  config :sequence, :validate => ["none", "epoch", "sequence"]
+
   def register
     @host = Socket.gethostname
+    @field_sequence = ecs_select[disabled: "clock", v1: "[event][sequence]"]
+    @field_host = ecs_select[disabled: "host", v1: "[host][name]"]
+    if sequence.nil? && ["epoch", "sequence"].include?(message)
+      logger.warn("message contains sequence type specification (epoch|sequence) for this purpose use the sequence option")
+    end
+    if ecs_compatibility == :disabled && @sequence.nil?
+      if %w(epoch sequence).include?(@message)
+        logger.debug("intercepting magic `message` to configure `sequence`: `#{@message}`")
+        @sequence, @message = @message, nil # legacy: intercept magic messages
+        deprecation_logger.deprecated("magic values of `message` to specify sequence type are deprecated; use separate `sequence` option instead.")
+      end
+    end
+    @sequence = "none" if @sequence.nil?
+    @sequence_selector = @sequence.to_sym
   end
 
   def run(queue)
-    sequence = 0
+    sequence_count = 0
 
     while !stop?
       start = Time.now
 
-      sequence += 1
-      event = generate_message(sequence)
+      sequence_count += 1
+      event = generate_message(sequence_count)
       decorate(event)
       queue << event
-      break if sequence == @count || stop?
+      break if sequence_count == @count || stop?
 
       sleep_for = @interval - (Time.now - start)
       Stud.stoppable_sleep(sleep_for) { stop? } if sleep_for > 0
     end
   end
 
-  def generate_message(sequence)
-    if @message == "epoch"
-      LogStash::Event.new("clock" => Time.now.to_i, "host" => @host)
-    elsif @message == "sequence"
-      LogStash::Event.new("clock" => sequence, "host" => @host)
-    else
-      LogStash::Event.new("message" => @message, "host" => @host)
+  def generate_message(sequence_count)
+    if @sequence_selector == :none
+      evt = LogStash::Event.new("message" => @message)
```

Comment:
sorry for the late :bee: but I think we might benefit from leveraging the event-support mixin here.
```suggestion
      evt = event_factory.new_event("message" => @message)
```
+ the obligatory mixin gem dependency [setup](https://github.com/logstash-plugins/logstash-codec-json/blob/740274885c9b35c6e4c0f33ab421d63632fa36f8/logstash-codec-json.gemspec#L27) + [require + include](https://github.com/logstash-plugins/logstash-codec-json/blob/740274885c9b35c6e4c0f33ab421d63632fa36f8/lib/logstash/codecs/json.rb#L30).